## Sampling Distribution Part-B

### D19001-Abhilasha Chourasia-Confidence Interval

##### Confidence Interval approach reports an interval (a,b) which can capture the population parameter with some confidence

##### Bootstrap to find the confidence interval estimate of the population parameter .
##### Use Monthly Income column from the dataset attrition.csv

In [1]:
#packages for Bootstrapping
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import math
import os
import seaborn as sns
import statsmodels.api as stm

#Setting the working directory
os.chdir("C:\\Users\\abhi3\\Downloads")

#Reading the data
attr = pd.read_csv("attrition.csv")

##### Q1.1.Consider entire variable as a population and calculate population mean

In [2]:
#reading the monthly income attribute from attrition.csv and calculating the mean for it
m_income=attr.MonthlyIncome
print('Mean of the population (MonthlyIncome): ',np.mean(m_income))

Mean of the population (MonthlyIncome):  6502.931292517007


##### Q1.2.Select sample 's' here (Xsam) of the size 200 from the population and calculate 95% of Confidence Interval estimate of the population mean.  

In [3]:
#sample from population of size 200 without replacement
Xsam=np.random.choice(m_income,200,replace=False)

In [4]:
#Calculating the value of intervals a,b
a=(np.mean(Xsam)-((1.96)*np.std(Xsam)/np.sqrt(len(Xsam))))  #z-score of 95%=1.96
b=(np.mean(Xsam)+((1.96)*np.std(Xsam)/np.sqrt(len(Xsam))))
print('The Confidence Interval for the sample:',a,',',b)
print('The mean for the Sample:',np.mean(Xsam))

The Confidence Interval for the sample: 6127.031820104587 , 7462.588179895414
The mean for the Sample: 6794.81


*Observation:* 

*i.Here 95% CI means we would like to create an interval (a,b) which will have the center sample mean such that this interval will be able to capture the population mean with probability 0.95*.

*ii.From above we see that with 95% CI(Confidence Interval,corresponding z score = 1.96), the mean score lies in the above interval. The value at 'a' is 2.5th percentile and value at 'b' is 97.5th percentile of the distribution of sample.*

*iii.Also if we evaluate, this interval captures the population mean as well i.e 6502.931292517007*

##### Q1.3.Repeat above 1000 times and check how many of those Confidence Interval captures the true population mean

In [6]:
#Below function creates 1000 such intervals on the sample and evaluates how many of these intervals 
#are capturing the sample mean with 95% CI.
def bootstrap_CI():
    true_count=0
    false_count=0
    for i in range(0,1000):
        D= np.random.choice(Xsam,200,replace=True)
        Xmean=int(np.mean(Xsam))
        
        a=int(np.mean(D)-(1.96*(np.std(D)/np.sqrt(200))))  #value of a i.e 2.5th percentile
        b=int(np.mean(D)+(1.96*(np.std(D)/np.sqrt(200))))  #value of b i.e 97.5th percentile
    
        
        if Xmean in range(a,b):                            #checks wheather the interval captures the mean
            true_count+=1
        else:
            false_count+=1
        
    print('Number of intervals which are able to capture the population parameter: ',true_count)   
    print('Number of intervals which are not able to capture the population parameter: ',false_count)

In [7]:
bootstrap_CI()

Number of intervals which are able to capture the population parameter:  944
Number of intervals which are not able to capture the population parameter:  56


*Observation:*

*i. We can see that with 95% Confidence Interval when we create 1000 such intervals, around 950 intervals are able to capture the sample mean such that these interval are able to capture population mean with probability 0.95.* 

### Wide vs Narrow Confidence Interval

##### Q2.Collect a sample 's' of size 200 from the population and calculate:
##### a)Confidence Interval=90%
##### b)Confidence Interval=99%

In [9]:
#CI-90%, z-score for 90% CI is 1.64
a=(np.mean(Xsam)-((1.64)*np.std(Xsam)/np.sqrt(len(Xsam))))
b=(np.mean(Xsam)+((1.64)*np.std(Xsam)/np.sqrt(len(Xsam))))
print('For CI 90%',a,',',b) 
print('Difference of the range',(b-a))

#CI-99%, z-score for 99% CI is 2.575
a=(np.mean(Xsam)-((2.575)*np.std(Xsam)/np.sqrt(len(Xsam))))
b=(np.mean(Xsam)+((2.575)*np.std(Xsam)/np.sqrt(len(Xsam))))
print('For CI 99%',a,',',b)
print('Difference of the range',(b-a))  #curve wider

For CI 90% 6236.0568290671035 , 7353.563170932897
Difference of the range 1117.5063418657937
For CI 99% 5917.499381004751 , 7672.12061899525
Difference of the range 1754.6212379904991


*Observation:*

*i.From above we can see that for 90% CI(zscore=1.64), the interval is narrower than that of with 99% CI(zscore=2.575) i.e interval with 99% CI is wider.(Clearly visible when difference of the interval range is taken)*

*ii.As the CI increses, the interval size increases i.e the interval is wider. The precision is less with high CI.As the CI increases more the intervals will represent the full sample range, hence the it is wider*

##### Q3.Collet 2 sample from the population S1 of size 200 and S2 of size 400 using both the samples. Calculate a 90% CI estimate of population mean, which one is narrower

In [10]:
#sample from population of size 200 without replacement
S1=np.random.choice(m_income,200,replace=False)  #S1 of size 200
#sample from population of size 200 without replacement
S2=np.random.choice(m_income,400,replace=False)  #S2 of size 400

In [11]:
#CI-90%, SIZE=200, zscore for 90% is 1.64
a1=(np.mean(S1)-((1.64)*np.std(S1)/np.sqrt(len(S1))))
b1=(np.mean(S1)+((1.64)*np.std(S1)/np.sqrt(len(S1))))

print('For CI 90%',a1,',',b1)
print('Difference',(b1-a1))

#CI-90%, SIZE=400
a2=(np.mean(S2)-((1.64)*np.std(S2)/np.sqrt(len(S2))))
b2=(np.mean(S2)+((1.64)*np.std(S2)/np.sqrt(len(S2))))

print('For CI 90%',a2,',',b2)
print('Difference',(b2-a2))
  #curve wider

For CI 90% 6024.2121428809705 , 7066.96785711903
Difference 1042.7557142380592
For CI 90% 6056.047454215063 , 6815.047545784936
Difference 759.0000915698729


*Observation:*
    
*i.From above we can see the interval of sample with more size(400) is narrower than the other(200)(clearly seen when difference of the interval is taken).*

*ii.Increasing the sample size decreases the width of CI, as it decreases the standard error.*
